### ISSUE TITLE
500 Error

### ISSUE THREAD

**Issue by MichelfrancisBustillos** [linuxserver/ifai#1103]
May be related to/same as [Issue 932](https://github.com/linuxserver/Heimdall/issues/932).

When adding an icon or updating apps list, site returns a 500 error. Attempted to update by pulling the latest version through Portainer and now 500 error is persistent. Am on the latest version

**Comment by rootedbox** [comment]
Same.. newest version.. I was able to add an item only one time and it required there be an icon attached.. but if I edit or try to add any more items I get 500.. once I get a 500 I have to restart the container. Logs from container below.

```
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    1000
User gid:    1000
-------------------------------------
cont-init: info: /etc/cont-init.d/10-adduser exited 0
cont-init: info: running /etc/cont-init.d/11-folders
cont-init: info: /etc/cont-init.d/11-folders exited 0
cont-init: info: running /etc/cont-init.d/12-samples
cont-init: info: /etc/cont-init.d/12-samples exited 0
cont-init: info: running /etc/cont-init.d/13-nginx
cont-init: info: /etc/cont-init.d/13-nginx exited 0
cont-init: info: running /etc/cont-init.d/14-php
cont-init: info: /etc/cont-init.d/14-php exited 0
cont-init: info: running /etc/cont-init.d/15-keygen
using keys found in /config/keys
cont-init: info: /etc/cont-init.d/15-keygen exited 0
cont-init: info: running /etc/cont-init.d/20-permissions
cont-init: info: /etc/cont-init.d/20-permissions exited 0
cont-init: info: running /etc/cont-init.d/50-config
cont-init: info: /etc/cont-init.d/50-config exited 0
cont-init: info: running /etc/cont-init.d/85-version-checks
cont-init: info: /etc/cont-init.d/85-version-checks exited 0
cont-init: info: running /etc/cont-init.d/99-custom-files
[custom-init] No custom files found, skipping...
cont-init: info: /etc/cont-init.d/99-custom-files exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-mods: starting
s6-rc: info: service init-mods successfully started
s6-rc: info: service init-mods-package-install: starting
s6-rc: info: service init-mods-package-install successfully started
s6-rc: info: service init-mods-end: starting
s6-rc: info: service init-mods-end successfully started
s6-rc: info: service init-services: starting
s6-rc: info: service init-services successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun cron (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
services-up: info: copying legacy longrun php-fpm (no readiness notification)
services-up: info: copying legacy longrun queue (no readiness notification)
s6-rc: info: service legacy-services successfully started
s6-rc: info: service 99-ci-service-check: starting
[ls.io-init] done.
s6-rc: info: service 99-ci-service-check successfully started
   GuzzleHttp\Exception\ConnectException 
  cURL error 6: Could not resolve host: appslist.heimdall.site (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://appslist.heimdall.site/list.json
  at /app/www/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:210
    206▕         }
    207▕ 
    208▕         // Create a connection exception if it was a specific error code.
    209▕         $error = isset($connectionErrors[$easy->errno])
  ➜ 210▕             ? new ConnectException($message, $easy->request, null, $ctx)
    211▕             : new RequestException($message, $easy->request, $easy->response, null, $ctx);
    212▕ 
    213▕         return P\Create::rejectionFor($error);
    214▕     }
      +13 vendor frames 
  14  /app/www/app/SupportedApps.php:157
      GuzzleHttp\Client::request()
  15  /app/www/app/Jobs/ProcessApps.php:42
      App\SupportedApps::getList()
   GuzzleHttp\Exception\ConnectException 
  cURL error 6: Could not resolve host: appslist.heimdall.site (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://appslist.heimdall.site/list.json
  at /app/www/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:210
    206▕         }
    207▕ 
    208▕         // Create a connection exception if it was a specific error code.
    209▕         $error = isset($connectionErrors[$easy->errno])
  ➜ 210▕             ? new ConnectException($message, $easy->request, null, $ctx)
    211▕             : new RequestException($message, $easy->request, $easy->response, null, $ctx);
    212▕ 
    213▕         return P\Create::rejectionFor($error);
    214▕     }
      +13 vendor frames 
  14  /app/www/app/SupportedApps.php:157
      GuzzleHttp\Client::request()
  15  /app/www/app/Jobs/ProcessApps.php:42
      App\SupportedApps::getList()
```

**Comment by artichoked1** [comment]
I'm having a similar problem on version 2.5.4. everything was working fine for a while but now when I try to add a new item to the dashboard, I get a 500 status code.
Edit: After updating to 2.5.5, all pages seem to return 500 status code.

**Comment by keriati** [comment]
> 

It looks like some network issues as the domain `appslist.heimdall.site` seems not to be reachable.

**Comment by MichelfrancisBustillos** [comment]
Update: I expended the storage space on the VM containing Heimdall and reload via Portainer. Issue now resolved.

**Comment by artichoked1** [comment]
Thank you!

**Comment by dyimo** [comment]
I think this problem is not fixed at all, the first time I use the environment, docker 2.5.6 version, when I start the container, wait for a while will be 500 error, check the logs found all appslist.heimdall.site can not be accessed

**Comment by j0nnymoe** [comment]
@dyimo then please open a new issue on the docker-heimdall repo and provide all the information requested.

**Comment by PedroBuffon** [comment]
still nothing of a fix? i can't even use the app , i try adding apps and there it is Gateway time-out 504, the only way to add something is to random put characters on the name and save, not adding the app itself, but that is a ugly way to add items, if there is no fix, saddly i'll be moving to homepage :(

**Comment by artichoked1** [comment]
Yeah, I'm still not having much luck either. It's pretty much unusable now. I might have to consider switching to Homer :(

**Comment by ezgoodnight** [comment]
I'll leave a comment because I had this issue and it does appear to be the same one as 932. It works normally now.

I tried setting up Radarr as my first app and repeatedly got the 500 Error.

I simply closed the window and reloaded it from the UNRAID docker menu. I did not reset or stop the container.

I tried adding Plex first this time. I used a URL with ".../web/index.html/" at the end and it added just fine with default settings.

Then I added Radarr and the 500 error I was getting with it went away. Now I can add whatever I want.

Maybe it could not write to the storage space on the first open. Whatever the case after minimal fiddling it did not work, and now it works. Perhaps a clean load and trying a few applications will resolve the problem for you.
